In [1]:
import pandas as pd
import numpy as np
import matplotlib as pyplot

In [2]:
df = pd.read_csv('Final_dataset.csv')

In [3]:
df.columns

Index(['Unnamed: 0', 'Date', 'STORE_NBR', 'LYLTY_CARD_NBR', 'TXN_ID',
       'PROD_NBR', 'product_name', 'Brand_name', 'packet_size', 'prod_qty',
       'total_sales', 'Year', 'month_num', 'month_name', 'weekday',
       'weekday_num', 'quarter', 'lifestage', 'premium_customer'],
      dtype='object')

In [4]:
df.drop('Unnamed: 0', axis = 1, inplace = True)

In [5]:
df = df.reset_index()

In [6]:
df.columns

Index(['index', 'Date', 'STORE_NBR', 'LYLTY_CARD_NBR', 'TXN_ID', 'PROD_NBR',
       'product_name', 'Brand_name', 'packet_size', 'prod_qty', 'total_sales',
       'Year', 'month_num', 'month_name', 'weekday', 'weekday_num', 'quarter',
       'lifestage', 'premium_customer'],
      dtype='object')

In [7]:
df.head(2)

index       Date  STORE_NBR  LYLTY_CARD_NBR  TXN_ID  PROD_NBR  \
0      0  5/20/2019          1            1343     383        61   
1      1  5/16/2019          4            4149    3333        16   

                          product_name Brand_name  packet_size  prod_qty  \
0     Smiths Crinkle Cut Chips Chicken     Smiths          170         2   
1  Smiths Crinkle Chips Salt & Vinegar     Smiths          330         1   

   total_sales  Year  month_num month_name   weekday  weekday_num  quarter  \
0          3.1  2019          5        May    Monday            2        2   
1          5.7  2019          5        May  Thursday            5        2   

                lifestage premium_customer  
0  MIDAGE SINGLES/COUPLES           Budget  
1  MIDAGE SINGLES/COUPLES           Budget

In [8]:
df.shape

(49770, 19)

In [9]:
x = pd.DataFrame(df.groupby(by = 'premium_customer')['index'].count())
x = x.reset_index()
x.rename(columns= {'index': 'Number of customers who buy chips'}, inplace = True)
x

premium_customer  Number of customers who buy chips
0           Budget                              17587
1       Mainstream                              19149
2          Premium                              13034

The mainstream customers have a large majority in the market who purchase chips as compared to the other categories Budget and Premium customers.
- There are 17,587 customers who buy chips products in Budget segment.
- There are 19,149 customers who buy chips products in Mainstream segment.
- There are 13,034 customers who buy chips products in Premium segment.

### Proportions

We can compute proportions of the customer segments to see what portion of the spending the customers spend on the chips-like category products and we can also see the Proportion of customers in each customer segment overall to compare against the 
mix of customers who purchase chip.So we need to compute two proportions:
- The proportion of the customers who purchase chips.
- The porportion of the spending on the chips category to overall all the categories.

Since, `Final_Dataset` includes all those customers who purchase chips products and we need to import our initial dataset `transaction_data` in which we have all the information about all the categories of the products which customers buys

In [12]:
df2 = pd.read_excel('transaction_data.xlsx')

In [13]:
df2.head(3)

Date  STORE_NBR  LYLTY_CARD_NBR  TXN_ID  PROD_NBR  \
0 2018-10-17          1            1000       1         5   
1 2019-05-14          1            1307     348        66   
2 2019-05-20          1            1343     383        61   

                            product_name  packet_size  prod_qty  total_sales  \
0        Natural Chip Compny SeaSalt175g          175         2          6.0   
1                  CCs Nacho Cheese 175g          175         3          6.3   
2  Smiths Crinkle Cut Chips Chicken 170g          170         2          2.9   

   Year  month_num month_name    weekday  weekday_num  quarter  
0  2018         10    October  Wednesday            4        4  
1  2019          5        May    Tuesday            3        2  
2  2019          5        May     Monday            2        2

In [14]:
df2.shape

(264836, 15)

In [15]:
# import the customers dataset
customer_df = pd.read_csv('purchase_behaviour.csv')

In [16]:
customer_df.head()

LYLTY_CARD_NBR               lifestage premium_customer
0            1000   YOUNG SINGLES/COUPLES          Premium
1            1002   YOUNG SINGLES/COUPLES       Mainstream
2            1003          YOUNG FAMILIES           Budget
3            1004   OLDER SINGLES/COUPLES       Mainstream
4            1005  MIDAGE SINGLES/COUPLES       Mainstream

In [17]:
customer_df.shape

(72637, 3)

We will merge the both datasets, `transaction dataset` which is stored in df2 and `purchase behaviour dataset` which is stored in customer_df. Since, we want all the records from the `transaction dataset` and all the records which match from the `transaction dataset` so will make a left join and will merge both datasets.

In [19]:
# merging both datasets df2 and customer_df
merged_df = pd.merge(df2, customer_df, how = 'left', on = 'LYLTY_CARD_NBR')

In [20]:
merged_df.drop_duplicates(inplace = True)

In [21]:
merged_df.head(3)

Date  STORE_NBR  LYLTY_CARD_NBR  TXN_ID  PROD_NBR  \
0 2018-10-17          1            1000       1         5   
1 2019-05-14          1            1307     348        66   
2 2019-05-20          1            1343     383        61   

                            product_name  packet_size  prod_qty  total_sales  \
0        Natural Chip Compny SeaSalt175g          175         2          6.0   
1                  CCs Nacho Cheese 175g          175         3          6.3   
2  Smiths Crinkle Cut Chips Chicken 170g          170         2          2.9   

   Year  month_num month_name    weekday  weekday_num  quarter  \
0  2018         10    October  Wednesday            4        4   
1  2019          5        May    Tuesday            3        2   
2  2019          5        May     Monday            2        2   

                lifestage premium_customer  
0   YOUNG SINGLES/COUPLES          Premium  
1  MIDAGE SINGLES/COUPLES           Budget  
2  MIDAGE SINGLES/COUPLES           Budget

In [22]:
merged_df.isnull().sum()

Date                0
STORE_NBR           0
LYLTY_CARD_NBR      0
TXN_ID              0
PROD_NBR            0
product_name        0
packet_size         0
prod_qty            0
total_sales         0
Year                0
month_num           0
month_name          0
weekday             0
weekday_num         0
quarter             0
lifestage           0
premium_customer    0
dtype: int64

Fortunately no missing values in our dataset.

Now we will see how many customers are there in each segement of customer as

In [25]:
y = pd.DataFrame(merged_df.groupby('premium_customer')['TXN_ID'].count()).reset_index()
y.rename(columns = {'TXN_ID': 'Total Number of customers'}, inplace = True)
y

premium_customer  Total Number of customers
0           Budget                      93157
1       Mainstream                     101988
2          Premium                      69690

In [26]:
# Let's merge x and y dataframes
segment_df = pd.merge(x,y, how  = 'outer', on = 'premium_customer')
segment_df

premium_customer  Number of customers who buy chips  \
0           Budget                              17587   
1       Mainstream                              19149   
2          Premium                              13034   

   Total Number of customers  
0                      93157  
1                     101988  
2                      69690

Now we can compute the proportion of customers who buy chips products by simply dividing the `Number of customers who buy chips` by `Total Number of customers` and multiply it by 100.


In [28]:
# computing propotion of each segement of the customers


In [29]:
segment_df['proportion(%)'] = (segment_df['Number of customers who buy chips'] / segment_df['Total Number of customers']) * 100
segment_df

premium_customer  Number of customers who buy chips  \
0           Budget                              17587   
1       Mainstream                              19149   
2          Premium                              13034   

   Total Number of customers  proportion(%)  
0                      93157      18.878882  
1                     101988      18.775738  
2                      69690      18.702827

- 18.87% of the Budget customers buy chips products.
- 18.77% of the Mainstream customers buy chips products.
- 18.70% of the Premium customers buy chips products.

Though, There is approximately 19% of the customers (budget segment) and 18% for both segments (Mainstream and Premium). Since the percentages are very close to each other which conludes that the proportion of the customers who buy chips for each segment is almost same.

___

Now we need to work on
**The porportion of the spending on the chips category to overall all the categories**. It will tell us that what proportion of their grocery is spent on the chips.

In [32]:
# total_sales of the chips for each segment of the customers.
z = pd.DataFrame(df.groupby('premium_customer')['total_sales'].sum()).reset_index()
z.rename(columns = {'total_sales': 'Sum of total sales of chips products'}, inplace = True)
z

premium_customer  Sum of total sales of chips products
0           Budget                              116978.3
1       Mainstream                              129502.6
2          Premium                               86485.1

In [33]:
# total sales of all products for each segment of the customers
j = pd.DataFrame(merged_df.groupby('premium_customer')['total_sales'].sum()).reset_index()
j.rename(columns = {'total_sales': 'Sum of total sales of all products'}, inplace = True)
j

premium_customer  Sum of total sales of all products
0           Budget                           676211.55
1       Mainstream                           750744.50
2          Premium                           507452.95

In [34]:
# merge both datasets z and j
merged_sales_df = pd.merge(z,j, how = 'outer', on = 'premium_customer')
merged_sales_df

premium_customer  Sum of total sales of chips products  \
0           Budget                              116978.3   
1       Mainstream                              129502.6   
2          Premium                               86485.1   

   Sum of total sales of all products  
0                           676211.55  
1                           750744.50  
2                           507452.95

In [35]:
# compute proportion of the spending on the chips
merged_sales_df['Proportion of spending on chips(%)'] = (merged_sales_df['Sum of total sales of chips products'] / merged_sales_df['Sum of total sales of all products']) * 100
merged_sales_df

premium_customer  Sum of total sales of chips products  \
0           Budget                              116978.3   
1       Mainstream                              129502.6   
2          Premium                               86485.1   

   Sum of total sales of all products  Proportion of spending on chips(%)  
0                           676211.55                           17.299069  
1                           750744.50                           17.249890  
2                           507452.95                           17.042979

We can see that the proportion of spending on the chips products is almost same for each segment of the customer and we can say that the customers spend 17% of their grocery spending on the chips products.

In other words we can say that the customers spend 17% (on chips products) of their proportion of the income which is spent on grocery items.
Also, we can say that the customers spend 83% on the other grocery items and 17% on the chips items.
___

In the excel file `final_dataset` we have pivot table on sheet named AVG price per unit. In that pivot table we computed the average price per unit sold by customers segments and the Mainstream midage and young singles / couples are more willing to pay more per 
packet of chips compared to their budget and premium counterpart. As we can see in that in the pivot table that the difference in the average prices per unit is not much larger for different customers segments so we can check that if this difference is statistically different by conducting two-sample independent t-test.

We will be conducting t test to see if there is any difference betweeen mainstream vs premium and budget midage and young singles and couples.
So in file we took only values which are extracted and can be seen in the file.
First group includes the average price per unit sold for mainstream segment young and midage singles and couples and simlarly in group2 we have combined elements of both segments budget and premium and same families are taken in this group. So let's start conducting t test by using python package scipys

In [38]:
# store the observations of average prices in two separate lists
grp_a = [3.627997211895, 3.71081193040575]
grp_b = [3.424607896, 3.415201005, 3.438390642, 3.374622222]

In [39]:
from scipy.stats import ttest_ind
t_test, p_value = ttest_ind(grp_a, grp_b, equal_var = False)

In [40]:
print(f"T-statistic: {t_test:.4f}, P-value: {p_value:.6f}")

T-statistic: 5.8735, P-value: 0.075929


P-value is not less than 0.05 (level of significance) so we condlude that the average price per unit sold for the mainstream midage and young singles and couples is not significantly higher than that of budget or premium, young and midage singles and couples.

It means that the average price per unit sold is same and are not significantly different.